### PR TITLE
Issue #3270853 by vnech: Redesign "add to Calendar" button on events

### PR DESCRIPTION
**Pull request by nechai**
## Problem
Move "add to Calendar" button to a link under the event date. This will improve the usability of the add to calendar feature and make it a more prominent functionality more integrated in the event enrolment process.

## Solution
Move "add to Calendar" button to a link under the event date

## Issue tracker
- https://www.drupal.org/project/social/issues/3270853
- https://getopensocial.atlassian.net/browse/YANG-6389

## Theme issue tracker
- [Social Base](https://www.drupal.org/project/socialbase/issues/3270772)
- [Social Blue](https://www.drupal.org/project/socialblue/issues/3270776)

## How to test
- [x] Deploy the instance with patches for themes
- [x] Login as SM
- [x] Enable "add to Calendar" feature - _/admin/config/opensocial/event-addtocal_
- [x] Visit any created event with future date

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
collapsed:
Screenshot 2022-04-08 at 18 35 28

expanded:
Screenshot 2022-04-08 at 18 35 35

## Release notes
*Redesign "add to Calendar" button on events.*

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by rollki** [comment]
LGTM, I have re-run tests.
@Ressinel Could you please check these changes?

**Review by rollki — APPROVED** [review]

**Comment by rollki** [comment]
Hi @nechai 
Could you please update screenshots in PR, seems it is looks a bit different at the moment.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by nechai** [comment]
@Ressinel Conflicts are fixed.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

File: modules/social_features/social_event/modules/social_event_addtocal/social_event_addtocal.module [hunk 1]
```diff
@@ -102,15 +182,20 @@ function _social_event_addtocal_get_links(NodeInterface $node) {
...
   return $links;
 }
+
+/**
+ * Implements theme_preprocess_HOOK() for "swiftmailer".
+ */
+function social_event_addtocal_preprocess_swiftmailer(array &$variables): void {
+  // Add library with styles for event.
+  $variables['#attached']['library'][] = 'social_event_addtocal/event_addtocal';
+  $variables['#attached']['library'][] = 'socialblue/dropdown';
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/modules/social_event_addtocal/social_event_addtocal.module]: I added check if theme `socialblue` is existed.

File: modules/social_features/social_event/modules/social_event_addtocal/social_event_addtocal.module [hunk 2]
```diff
@@ -41,25 +44,102 @@ function social_event_addtocal_node_view(array &$build, NodeInterface $node, Ent
...
+    // Add styles for the 'Add to calendar' links.
+    $build['#attached']['library'][] = 'social_event_addtocal/event_addtocal';
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK() for "input--button--split.html.twig".
+ */
+function social_event_addtocal_preprocess_input__button__split(array &$variables): void {
+  if (!empty($variables['element']['#value']) && $variables['element']['#value'] == 'Add to calendar') {
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/modules/social_event_addtocal/social_event_addtocal.module]: @Ressinel Fixed the issue (added checking by class attribute existing)

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Comment by tbsiqueira** [comment]
Hi @nechai, can you please resolve the conflict on this PR? Thank you!

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2824-0o5hvm9avutckp6tslfyzvtjpq0extdp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6238e37ac49d661fdca305b0

**Review by Ressinel — APPROVED** [review]
LGTM.

**Comment by Ressinel** [comment]
🍒-picked to `11.3.x`

**Pull request merged**